### DISCUSSION TITLE
Hyper-SDXL Preset

### DISCUSSION THREAD

**Discussion by tbuyle** [lllyasviel/Fooocus#2794] [Show and tell]
I wanted to test the new Hyper-SD and created a simple preset to use it.

[hypersd.json](https://github.com/lllyasviel/Fooocus/files/15072765/hypersd.json)

I choose to use the 4 steps Lora but similar settings should work with other step counts.

Some parameters may still require tweaking, it's really a quick and dirty first version.

![image](https://github.com/lllyasviel/Fooocus/assets/467350/e2047256-1f4b-437d-ae0d-5e14e7aec180)

**Comment by xhoxye** [comment]
```
    "default_sampler": "ddim",
    "default_scheduler": "sgm_uniform",
```
Is this the best combination?

**Reply by tbuyle** [reply]
Not sure. But it's the one I found in the ComfyUI workflow in the HF Repo : https://huggingface.co/ByteDance/Hyper-SD/blob/main/comfyui/Hyper-SDXL-Nsteps-lora-workflow.json

**Comment by xhoxye** [comment]
My test results, using the combination below is better.
Fooocus 2.3.1, Juggernaut
euler_a + karras，CFG=1.0
4 steps Lora, steps=8 , 
2 steps Lora, steps=5 , 

**test**
Fooocus v2.3.1
juggernautXL_version6Rundiffusion.safetensors
styles = none
<details>
<summary>A female magician wearing a magic hat,Sitting at the street coffee table drinking coffee</summary>
 
![light](https://github.com/lllyasviel/Fooocus/assets/129571231/4faaa517-1bd6-4f57-a2cc-2458edd587f5)
![2024-04-27_08-39-42_7760](https://github.com/lllyasviel/Fooocus/assets/129571231/394b8ac4-eb59-43d2-9677-e30ca77251d1)

![euler_a-8](https://github.com/lllyasviel/Fooocus/assets/129571231/d2bc2dd3-aada-4262-90b5-44b790a32f6f)
![2024-04-27_08-38-37_7855](https://github.com/lllyasviel/Fooocus/assets/129571231/7db8e006-d46c-4a81-a94e-3b76c7b4045e)

![ddim-4](https://github.com/lllyasviel/Fooocus/assets/129571231/669a1674-457b-4927-b23d-bd6a2fda7006)
![2024-04-27_08-37-24_5784](https://github.com/lllyasviel/Fooocus/assets/129571231/b55edda3-9cf4-458d-9f20-a10d8d869014)

 
</details>

**Comment by tbuyle** [comment]
I think we'll have to test en tinker with params before finding the best ones. Simply update the preset as per your preferences.

There are some interesting suggestions on these reddit posts :
- https://www.reddit.com/r/StableDiffusion/comments/1caxefs/hyper_sdxl_best_settings_and_my_assessment_of_it/
- https://www.reddit.com/r/StableDiffusion/comments/1caalgo/from_the_creators_of_sdxl_lightning_introduce/

**Comment by mashb1t** [comment]
@tbuyle Fooocus sadly currently doesn't support TCDScheduler, which is why it can't use Hyper-SD or TCD to full potential.

**Reply by wangfeng35** [reply]
Correct me if I get it wrong.
In https://huggingface.co/ByteDance/Hyper-SD , only [Unified LoRA](https://huggingface.co/ByteDance/Hyper-SD#unified-lora-support-1-to-8-steps-inference) using the TCDScheduler, while [2-Steps, 4-Steps, 8-steps LoRA](https://huggingface.co/ByteDance/Hyper-SD#2-steps-4-steps-8-steps-lora) are using DDIMScheduler. So is it possibile to using Hyper-SD 2-Steps, 4-Steps, 8-steps LoRA in Fooocus to full potential?

**Reply by mashb1t** [reply]
@wangfeng35 depends on your definition of full potential. I'll check this out today in the evening (in 6 hours) and give the LoRAs a try tro evaluate if we should add another performance mode!

**Reply by wangfeng35** [reply]
Thanks! You're making Fooocus awesome!

**Reply by mashb1t** [reply]
implemented as WIP for evaluation, see https://github.com/lllyasviel/Fooocus/pull/2812

**Comment by mashb1t** [comment]
The poll for Lightning vs Hyper-SD (both 4step LoRA) is up now and you can participate here: https://github.com/lllyasviel/Fooocus/discussions/2813
